[4.x] `OidcEncryption#generateMasterPassword` is failing on Windows and generate several tests failure

## Environment Details
* Helidon Version: 4.x
* Helidon SE or Helidon MP
* JDK version: 21
* OS: Windows
* Docker version (if applicable):

----------

## Problem Description

Windows file system does not implement POSIX standards.

## Steps to reproduce

run test

```bash
cd tests/integration/oidc
mvn install
...
[ERROR] io.helidon.tests.integration.oidc.IdTokenIT.testAuthenticationWithoutIdToken(WebTarget) -- Time elapsed: 1.490 s <<< ERROR!
java.lang.UnsupportedOperationException
        at java.base/java.nio.file.Files.setPosixFilePermissions(Files.java:2168)
        at io.helidon.security.providers.oidc.common.OidcEncryption.generateMasterPassword(OidcEncryption.java:85)
        at io.helidon.security.providers.oidc.common.OidcEncryption.create(OidcEncryption.java:53)
        at io.helidon.security.providers.oidc.common.OidcCookieHandler.<init>(OidcCookieHandler.java:90)
        at io.helidon.security.providers.oidc.common.OidcCookieHandler$Builder.build(OidcCookieHandler.java:217)
        at io.helidon.security.providers.oidc.common.OidcConfig.<init>(OidcConfig.java:425)
        at io.helidon.security.providers.oidc.common.OidcConfig$Builder.build(OidcConfig.java:986)
        at io.helidon.security.providers.oidc.common.OidcConfig.create(OidcConfig.java:456)
        at io.helidon.security.providers.oidc.OidcProvider$Builder.config(OidcProvider.java:332)
        at io.helidon.security.providers.oidc.OidcProvider.create(OidcProvider.java:105)
        at io.helidon.security.providers.oidc.OidcProviderService.providerInstance(OidcProviderService.java:45)
        at io.helidon.security.Security$Builder.providerFromConfig(Security.java:1124)
        at io.helidon.security.Security$Builder.lambda$fromConfig$1(Security.java:979)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at io.helidon.security.Security$Builder.lambda$fromConfig$2(Security.java:979)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.helidon.common.mapper.OptionalValue.ifPresent(OptionalValue.java:188)
        at io.helidon.security.Security$Builder.fromConfig(Security.java:978)
        at io.helidon.security.Security$Builder.config(Security.java:838)
        at io.helidon.microprofile.security.SecurityCdiExtension.configure(SecurityCdiExtension.java:95)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
        at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:85)
        at org.jboss.weld.injection.MethodInvocationStrategy$SimpleMethodInvocationStrategy.invoke(MethodInvocationStrategy.java:168)
        at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
        at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:126)
        at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
        at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
        at jakarta.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:142)
        at org.jboss.weld.util.Observers.notify(Observers.java:166)
        at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
        at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
        at org.jboss.weld.event.EventImpl.fire(EventImpl.java:96)
        at io.helidon.microprofile.cdi.HelidonContainerImpl.doStart(HelidonContainerImpl.java:306)
        at io.helidon.common.context.Contexts.runInContext(Contexts.java:137)
        at io.helidon.microprofile.cdi.HelidonContainerImpl.start(HelidonContainerImpl.java:260)
        at io.helidon.microprofile.cdi.HelidonContainerInitializer.initialize(HelidonContainerInitializer.java:154)
        at io.helidon.microprofile.testing.junit5.HelidonJunitExtension.startContainer(HelidonJunitExtension.java:364)
        at io.helidon.microprofile.testing.junit5.HelidonJunitExtension.beforeEach(HelidonJunitExtension.java:214)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeEachCallbacks$2(TestMethodTestDescriptor.java:166)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeBeforeMethodsOrCallbacksUntilExceptionOccurs$6(TestMethodTestDescriptor.java:202)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeMethodsOrCallbacksUntilExceptionOccurs(TestMethodTestDescriptor.java:202)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeBeforeEachCallbacks(TestMethodTestDescriptor.java:165)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:132)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
        at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
        at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
```

--- a/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
+++ b/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -82,7 +82,9 @@ final class OidcEncryption {
             String password = UUID.randomUUID().toString();
             try {
                 Files.writeString(path, password, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW);
-                Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
+                if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
+                    Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
+                }
             } catch (IOException e) {
                 throw new SecurityException("Failed to create OIDC secret " + path.toAbsolutePath(), e);
             }

--- a/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
+++ b/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
@@ -88,9 +88,9 @@ final class OidcEncryption {
                 } else {
                     File file = path.toFile();
                     boolean readable = file.setReadable(true, true);
-                    boolean executable = file.setExecutable(true, true);
-                    if (!readable || !executable) {
-                        LOGGER.log(Level.DEBUG, "Could not set file permission to " + file.toPath());
+                    boolean executable = file.setWritable(true, true);
+                    if ((!readable || !executable) && LOGGER.isLoggable(Level.DEBUG)) {
+                        LOGGER.log(Level.DEBUG, "Could not set file permission to " + path);
                     }
                 }
             } catch (IOException e) {

--- a/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
+++ b/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
@@ -16,7 +16,6 @@
 
 package io.helidon.security.providers.oidc.common;
 
-import java.io.File;
 import java.io.IOException;
 import java.lang.System.Logger.Level;
 import java.nio.charset.StandardCharsets;
@@ -85,13 +84,6 @@ final class OidcEncryption {
                 Files.writeString(path, password, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW);
                 if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
                     Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
-                } else {
-                    File file = path.toFile();
-                    boolean readable = file.setReadable(true, true);
-                    boolean executable = file.setWritable(true, true);
-                    if ((!readable || !executable) && LOGGER.isLoggable(Level.DEBUG)) {
-                        LOGGER.log(Level.DEBUG, "Could not set file permission to " + path);
-                    }
                 }
             } catch (IOException e) {
                 throw new SecurityException("Failed to create OIDC secret " + path.toAbsolutePath(), e);

--- a/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
+++ b/security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/OidcEncryption.java
@@ -16,6 +16,7 @@
 
 package io.helidon.security.providers.oidc.common;
 
+import java.io.File;
 import java.io.IOException;
 import java.lang.System.Logger.Level;
 import java.nio.charset.StandardCharsets;
@@ -84,6 +85,13 @@ final class OidcEncryption {
                 Files.writeString(path, password, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW);
                 if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
                     Files.setPosixFilePermissions(path, Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE));
+                } else {
+                    File file = path.toFile();
+                    boolean readable = file.setReadable(true, true);
+                    boolean executable = file.setExecutable(true, true);
+                    if (!readable || !executable) {
+                        LOGGER.log(Level.DEBUG, "Could not set file permission to " + file.toPath());
+                    }
                 }
             } catch (IOException e) {
                 throw new SecurityException("Failed to create OIDC secret " + path.toAbsolutePath(), e);

